Refactor Grunt targets.

XhmikosR opened the pull request:
* Add lib to JSHint
* Run htmlint own task by default
* Remove non existent `watch` target
* Add a test target (alias for default atm)

jzaefferer commented:
I'm not sure why you had the htmllint target removed, that's pretty important. I've applied the other changes separately. Thanks!

XhmikosR commented:
Because you run those tests in nodeunit so they are just duplicate...

XhmikosR commented:
Also, you didn't cherry pick my patch so I don't show as the author. That isn't cool.

jzaefferer commented:
I didn't cherry-pick since you removed half the file. I run that target manually, to actually test the grunt task, which the unit tests don't cover.

XhmikosR commented:
But you still do **not** run the htmllint target...

XhmikosR commented:
If you had just cherry picked **my patch**  https://github.com/jzaefferer/grunt-html/commit/5c60a4125541a3b7b9f2ce74634f7d0be52562d7, things would be OK.

But anyway, your project, your call.

jzaefferer commented:
I run the htmllint target **manually**, which helps to verify that console reporter.

XhmikosR commented:
You should add it to the test target. That is the point of a CI system like Travis.

jzaefferer commented:
How would adding a failing target help?

XhmikosR commented:
In my patch, I added `force` for the failing target.  https://github.com/jzaefferer/grunt-html/commit/5c60a4125541a3b7b9f2ce74634f7d0be52562d7#diff-35b4a816e0441e6a375cd925af50752cR11 and added the ignores for the other one.
